lib/loan_creator/common.rb

```diff
@@ -194,5 +198,31 @@ def term_zero_date
     def term_zero?
       interests_start_date && interests_start_date < term_zero_date
     end
+
+    def compute_last_term
+      @crd_beginning_of_period = @crd_end_of_period
+      @crd_end_of_period                  = bigd('0')
+      @due_interests_beginning_of_period  = @due_interests_end_of_period
+      @period_theoric_interests           = @due_interests_beginning_of_period + compute_period_generated_interests
+
+      @delta_interests = @period_theoric_interests - @period_theoric_interests.round(2)
+      @accrued_delta_interests += @delta_interests
+      @amount_to_add = bigd(@accrued_delta_interests.truncate(2))
+      @accrued_delta_interests -= @amount_to_add
+
+      @period_interests                   = @period_theoric_interests.round(2) + @amount_to_add
+      @due_interests_end_of_period        = 0
+      @period_capital                     = @crd_beginning_of_period
+      @total_paid_capital_end_of_period   += @period_capital
+      @total_paid_interests_end_of_period += @period_interests
+      @period_amount_to_pay               = @period_capital + @period_interests
+      # handled in Timetable.<<
+      @due_on = nil
+      if @index.nil?
+        @index = @starting_index
+      else
+        @index += 1
+      end
+    end
```

Comment:
De ce que m'a expliqué quentin, là tu as fais l'implémentation pour le dernier term, ce qui correspond à ton besoin actuel avec le in fine. Mais cela ne devrait-il pas être géré pour tous les terms dans le cas général ? Les intérêts capitalisés devraient être pris en compte dès les premiers remboursements de capital si il y en a, et non uniquement la dernière échéance.

Cela ne fait pas partie de ton besoin mais j'ai l'impression que le résultat actuel est incorrect pour ces cas.

Ça a une importance pour le code aussi car il ne faudrait pas rajouter cette méthode `compute_last_term` pour les échéanciers linear et standard, mais le gérer dans leur méthode `compute_current_term`.

Reply:
Je pense qu'une seule methode compute last term est une amelioration comparé aux executions dans compute_current_term si @last_term

Reply:
"là tu as fais l'implémentation pour le dernier term" -> non j'ai adapté tous les terms avec compute_period_generated_interests